### ISSUE TITLE
Many outdated things in the code requires instant follow up and update!!!!!!!!!!!!!!!!!!!!!!!!!!

### ISSUE THREAD

**Issue by TheShortKing69** [nicknochnack/TFODCourse#153]
Hi there, as someone who happens to start learning TFOD in 2023, I found out there is a lot of outdated coding in both image processing and training file. Please do consider to run them and update any outdated things to prevent error. It is a bit misguiding and confusing for many of us at the moment~!!!

PLEASE Update~!!! orz

**Comment by TheShortKing69** [comment]
Problem encountered 1:
File name: 2. Training and Detection-checkpoint.ipynb
Section: 6. Train the model
Line : !{command}
Error: AttributeError: module 'tensorflow.python.ops.control_flow_ops' has no attribute 'case'
Source of problem: C:\Users\HP\Documents\TFODCourse\tfod\Lib\site-packages\tf_slim-1.1.0-py3.10.egg\tf_slim\data\tfexample_decoder.py
At line 473 : image = control_flow_ops.case(

Is anyone facing any issue with this currently? I tried to run the original jupyter book Nick wrote and it shows the same error.

**Comment by abrar-nazib** [comment]
Yeah, I'm facing the same issue. Did you get any solution to this?

**Comment by TheShortKing69** [comment]
> Yeah, I'm facing the same issue. Did you get any solution to this?

I just went to the TensorFlow documentation and find out about this "control_flow_case". Maybe you could give a shot by importing "control_flow_case":
from tensorflow.python.ops import control_flow_case
in your tfexample_decoder.py

Currently I am having a new issue after installing CUDA cudann in my lappy. I can't open jupyter notebook because of new errors.
New Error:  
ImportError: cannot import name 'HashTrieMap' from 'rpds' (C:\Users\HP\Documents\TFODCourse\tfod\Lib\site-packages\rpds\__init__.py)
shit is getting tough every time I close and reopen it lmao. Kinda want to give up and just stick to Colab forever. :melting_face:

**Comment by BMugo84** [comment]
> Problem encountered 1: File name: 2. Training and Detection-checkpoint.ipynb Section: 6. Train the model Line : !{command} Error: AttributeError: module 'tensorflow.python.ops.control_flow_ops' has no attribute 'case' Source of problem: C:\Users\HP\Documents\TFODCourse\tfod\Lib\site-packages\tf_slim-1.1.0-py3.10.egg\tf_slim\data\tfexample_decoder.py At line 473 : image = control_flow_ops.case(
> 
> Is anyone facing any issue with this currently? I tried to run the original jupyter book Nick wrote and it shows the same error.

Hey there. I am also facing this issue in collab. And collab cant install tensorflow<2.12. 
I was able to bypass the cuDNN not registered by adding this line and switching theruntime from gpu to TPU:

```
# Avoid OOM errors by setting GPU Memory Consumption Growth
import tensorflow as tf
gpus = tf.config.experimental.list_physical_devices('GPU')
for gpu in gpus:
    tf.config.experimental.set_memory_growth(gpu, True)
```
currently using tensorflow==2.13.0

**Comment by BMugo84** [comment]
too many errors occuring from the deprecated model used. try upgrading to the newest version from here 
https://github.com/tensorflow/models/blob/master/official/vision/MODEL_GARDEN.md

as seen from [here](https://discuss.tensorflow.org/t/the-runtime-issue-on-google-colab-when-using-the-tensorflow-object-detection-model/17520)

**Comment by BMugo84** [comment]
there are two choices, either downgrade all your packages to match his, or reconfigure his code to match the latest tf model zoo/vision

**Comment by sujirou** [comment]
> there are two choices, either downgrade all your packages to match his, or reconfigure his code to match the latest tf model zoo/vision

did you downgrade? or found a workaround? downgrading seems like it's in order of magnitude more tedious than actually learning the lesson iteslf. how'd it work out for you?
